### ISSUE TITLE
1.1.0 crash on MacOS (M2)

### ISSUE THREAD

**Issue by rstz** [tensorflow/decision-forests#152]
Hi, in my Mac (M2) VSCode crashes always when run "import tensorflow_decision_forests as tfdf". I already reinstalled tensorflow (2.11.0), tensorflow-macos (2.11.0), tensorflow-metal (0.7.0) and tensorflow_decision_forests (1.1.0). Any clue? 

"Canceled future for execute_request message before replies were done
The Kernel crashed while executing code in the the current cell or a previous cell. Please review the code in the cell(s) to identify a possible cause of the failure. Click [here](https://aka.ms/vscodeJupyterKernelCrash) for more info. View Jupyter log for further details."

_Originally posted by @mtcdultra in https://github.com/tensorflow/decision-forests/issues/16#issuecomment-1373905354_

**Comment by rstz** [comment]
Hi Marcos, thank you for your report. Can you please provide more information about the python environment you're using? It sounds like you're in VSCode with an IPython / Jupyter environment, but I want to make sure in order to reproduce this

**Comment by mtcdultra** [comment]
Hi Richard, yes, I´m using VSCode with Jupyter environment. And on Mac terminal, tried to use iPtyhon directly and crashed again: "zsh: illegal hardware instruction  ipython".

Below is the code that I've tried run:

import pandas as pd
import tensorflow as tf
import tensorflow_decision_forests as tfdf

# Load a dataset into a Pandas Dataframe.
dataset_df = pd.read_csv("SP202102.csv", encoding='ISO-8859-1')

# Show the first three examples.
print(dataset_df.head(3))

# Convert the pandas dataframe into a tf dataset.
dataset_tf = tfdf.keras.pd_dataframe_to_tf_dataset(dataset_df, label="species")

# Train the Random Forest
model = tfdf.keras.RandomForestModel(compute_oob_variable_importances=True)
model.fit(x=dataset_tf)

**Comment by arnavrneo** [comment]
Hi @mtcdultra,

Can you try downgrading the `tensorflow-metal` to ver `0.5.0` or lower?

**Comment by rstz** [comment]
Hi, sorry for taking a bit of time to get to this. It looks like `tensorflow-metal` is incompatible with TF-DF at this time. This is on our Todo list, but we will need additional investigation on our side. 

@mtcdultra Can you confirm that it works if you uninstall `tensorflow-metal`?

**Comment by Arnold1** [comment]
does this also apply to m1?

**Comment by rstz** [comment]
Yes, TF-DF is not compatible with `tensorflow-metal` on M1 processors. This is a tricky issue to solve AFAICT: Normal tensorflow and tensorflow-metal have different interfaces, but TF-DF only ships a single binary per OS / processor / python version on Pypi. Any ideas are very welcome!
